File: server/bcephal-reco/src/main/java/com/moriset/bcephal/reconciliation/domain/ReconciliationData.java
/**
 * 
 */
package com.moriset.bcephal.reconciliation.domain;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

import com.moriset.bcephal.domain.dimension.Attribute;
import com.moriset.bcephal.domain.dimension.AttributeValue;
import com.moriset.bcephal.domain.dimension.EnrichmentValue;
import com.moriset.bcephal.grid.domain.GrilleDataFilter;

import lombok.Data;

/**
 * @author Joseph Wambo
 *
 */
@Data
public class ReconciliationData {

	private Long reconciliationId;
	private List<Long> leftids;
	private List<Long> rightids;
	
	private boolean selectAllRowsInLeftGrid;
	private GrilleDataFilter leftBrowserDataFilter;
	
	private boolean selectAllRowsInRightGrid;
	private GrilleDataFilter rightBrowserDataFilter;
	
	private boolean useDebitCredit;
	
	private Long recoTypeId;
	
	private Long recoSequenceId;
	
	private Long leftMeasureId;	
	private Long rigthMeasureId;

	private boolean performPartialReco;
    private boolean allowPartialReco;
    private Long partialRecoAttributeId;
    private Long partialRecoSequenceId;
    private Long reconciliatedMeasureId;
    private Long remainningMeasureId;
    public List<PartialRecoItem> partialRecoItems;

    private boolean allowFreeze;
    private Long freezeAttributeId;
    private Long freezeSequenceId;

    private boolean performNeutralization;
    private boolean allowNeutralization;
    private Long neutralizationAttributeId;
    private Long neutralizationSequenceId;
    private String neutralizationValue;
	
    private Long noteAttributeId;
	
	private Long writeOffMeasureId;
	private BigDecimal writeOffAmount;
	private BigDecimal leftAmount;
	private BigDecimal rigthAmount;	
	private BigDecimal balanceAmount;
	private BigDecimal reconciliatedAmount;
	private List<WriteOffField> writeOffFields;
	
	private boolean addRecoDate;	
	private boolean addUser;	
	private boolean addNote;	
	private boolean mandatoryNote;	
	private String note;	
	private boolean addAutomaticManual;	
	private Long recoDateId;
	
	private Attribute autoManualAttribute;
	private AttributeValue automaticValue;
	private AttributeValue manualValue;
	private Attribute userAttribute;
	
	private Integer defaultRecoValue = null;
	
	private String primaryView;
	private String SecondaryView;
	private List<Long> primaryids;
	private List<Long> secondaryids;
	
	private List<EnrichmentValue> enrichmentItemDatas;
	
	
	private boolean allowReconciliatedAmountLog;	
	private Long reconciliatedAmountLogGridId;
	
	
	public ReconciliationData() {
		leftids = new ArrayList<Long>(0);
		rightids = new ArrayList<Long>(0);
		writeOffAmount = BigDecimal.ZERO;
		leftAmount = BigDecimal.ZERO;
		rigthAmount = BigDecimal.ZERO;	
		balanceAmount = BigDecimal.ZERO;
		reconciliatedAmount = BigDecimal.ZERO;
		primaryids = new ArrayList<Long>(0);
		secondaryids = new ArrayList<Long>(0);
		writeOffFields = new ArrayList<WriteOffField>(0);
		enrichmentItemDatas = new ArrayList<EnrichmentValue>(0);
		partialRecoItems = new ArrayList<PartialRecoItem>(0);
	}
}

File: client/bcephal-client-model/Reconciliation/ReconciliationData.cs
using Bcephal.Models.Dimensions;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Reconciliation
{
    public class ReconciliationData
    {
        public long? ReconciliationId { get; set; }
        public ObservableCollection<long?> Leftids { get; set; }
        public ObservableCollection<long?> Rightids { get; set; }

        public long? RecoTypeId { get; set; }
        public long? RecoSequenceId { get; set; }

        public long? LeftMeasureId { get; set; }
        public long? RigthMeasureId { get; set; }

        public bool PerformPartialReco { get; set; }
        public bool AllowPartialReco { get; set; }
        public long? PartialRecoAttributeId { get; set; }
        public long? PartialRecoSequenceId { get; set; }
        public long? ReconciliatedMeasureId { get; set; }
        public long? RemainningMeasureId { get; set; }

        public ObservableCollection<PartialRecoItem> PartialRecoItems { get; set; }

        public bool AllowFreeze { get; set; }
        public long? FreezeAttributeId { get; set; }
        public long? FreezeSequenceId { get; set; }

        public bool AllowNeutralization { get; set; }
        public long? NeutralizationAttributeId { get; set; }
        public long? NeutralizationSequenceId { get; set; }
        public string NeutralizationValue { get; set; }

        public long? WriteOffMeasureId { get; set; }
        public decimal WriteOffAmount { get; set; }
        public decimal LeftAmount { get; set; }
        public decimal RigthAmount { get; set; }
        public decimal BalanceAmount { get; set; }
        public ObservableCollection<WriteOffField> WriteOffFields { get; set; }

        public ObservableCollection<ReconciliationModelEnrichment> EnrichmentItemDatas { get; set; }

        public bool AddRecoDate { get; set; }
        public bool AddUser { get; set; }
        public bool AddNote { get; set; }
        public bool MandatoryNote { get; set; }
        public string Note { get; set; }
        public bool AddAutomaticManual { get; set; }
        public long? RecoDateId { get; set; }

        public ReconciliationData()
        {
            Leftids = new ObservableCollection<long?>();
            Rightids = new ObservableCollection<long?>();
            WriteOffAmount = 0;
            LeftAmount = 0;
            RigthAmount = 0;
            BalanceAmount = 0;
            EnrichmentItemDatas = new ObservableCollection<ReconciliationModelEnrichment>();
        }

    }
}
